ovs-p4rt/sidecar/testing/prepare_l2_to_tunnel_test.cc

```diff
@@ -0,0 +1,198 @@
+
+// TODO(derek): How stable are the table and action IDs?
+constexpr uint32_t kTableIdV4 = 43337754U;
+constexpr uint32_t kActionIdV4 = 23805991U;
+
+void get_v4_fdb_info(mac_learning_info& fdb_info) {
+  memcpy(fdb_info.mac_addr, kMacAddrV4, sizeof(fdb_info.mac_addr));
+  fdb_info.tnl_info.remote_ip.family = AF_INET;
+  fdb_info.tnl_info.remote_ip.ip.v4addr.s_addr = kIpAddrV4;
+}
+
+TEST_F(PrepareL2ToTunnelTest, GetL2ToTunnelV4TableEntry) {
+  // Arrange
+  struct mac_learning_info fdb_info = {0};
+  get_v4_fdb_info(fdb_info);
+
+  // Act
+  ::p4::v1::TableEntry table_entry;
+  DiagDetail detail;
+  PrepareL2ToTunnelV4(&table_entry, fdb_info, p4info, true, detail);
+#ifdef DUMP_JSON
+  DumpMessageAsJson(table_entry);
+#endif
+
+  // Assert
+  ASSERT_EQ(table_entry.table_id(), kTableIdV4);
+
+  // match: mac_addr
+  ASSERT_EQ(table_entry.match_size(), 1);
+
+  auto match = table_entry.match()[0];
+  ASSERT_EQ(match.field_id(), 1);
+
+  ASSERT_TRUE(match.has_exact());
+
+  auto match_value = match.exact().value();
+  ASSERT_EQ(match_value.size(), sizeof(kMacAddrV4));
+
+  for (int i = 0; i < sizeof(kMacAddrV4); i++) {
+    ASSERT_EQ(uint8_t(match_value[i]), kMacAddrV4[i]);
+  }
+
+  // action: set_tunnel_v4(dst_addr)
+  ASSERT_TRUE(table_entry.has_action());
+
+  auto table_action = table_entry.action();
+  auto action = table_action.action();
+  ASSERT_EQ(action.action_id(), kActionIdV4);
+
+  ASSERT_EQ(action.params_size(), 1);
+
+  auto param = action.params()[0];
+  ASSERT_EQ(param.param_id(), 1);
+
+  auto param_value = param.value();
+  ASSERT_EQ(param_value.size(), 4);
+
+  auto word_value = ntohl(DecodeWordValue(param_value));
+  ASSERT_EQ(word_value, kIpAddrV4);
+}
+
+//----------------------------------------------------------------------
+// GetL2ToTunnelV6TableEntry
+//----------------------------------------------------------------------
+constexpr uint8_t kMacAddrV6[] = {0xb, 0xe, 0xe, 0xb, 0xe, 0xe};
+constexpr uint32_t kIpAddrV6[] = {0, 66, 129, 512};
+constexpr int kIpAddrV6Len = sizeof(kIpAddrV6) / sizeof(kIpAddrV6[0]);
+
+// TODO(derek): How stable are the table and action IDs?
+constexpr uint32_t kTableIdV6 = 36796227U;
+constexpr uint32_t kActionIdV6 = 23953453U;
+
+void get_v6_fdb_info(mac_learning_info& fdb_info) {
+  memcpy(fdb_info.mac_addr, kMacAddrV6, sizeof(fdb_info.mac_addr));
+  fdb_info.tnl_info.remote_ip.family = AF_INET6;
+  for (int i = 0; i < kIpAddrV6Len; i++) {
+    fdb_info.tnl_info.remote_ip.ip.v6addr.__in6_u.__u6_addr32[i] = kIpAddrV6[i];
+  }
+}
+
+TEST_F(PrepareL2ToTunnelTest, GetL2ToTunnelV6TableEntry) {
+  // Arrange
+  struct mac_learning_info fdb_info = {0};
+  get_v6_fdb_info(fdb_info);
+
+  // Act
+  ::p4::v1::TableEntry table_entry;
+  DiagDetail detail;
+  PrepareL2ToTunnelV6(&table_entry, fdb_info, p4info, true, detail);
+#ifdef DUMP_JSON
+  DumpMessageAsJson(table_entry);
+#endif
+
+  // Assert
+  ASSERT_EQ(table_entry.table_id(), kTableIdV6);
```

Comment:
Should this PR also include a p4info.txt file?
which has kTableIdV4, kActionIdV4, kTableIdV6 and kActionIdV6

Reply:
I'd like to include it. I was concerned that it might be considered proprietary.